### ISSUE TITLE
[@nativescript/email] Crash on iOS >= 15.0

### ISSUE THREAD

**Issue by benedictstrube** [NativeScript/plugins#182]
On iOS Versions >= 15.0, our App immediately crashes without logs when invoking the plugin in the latest version in a NS8 project. Everything went fine until we updated the iOS version. To overcome these crashes, we tried mailto links instead. However, these also crash our app (at least in a production build). So I am not too sure, whether this is an OS or plugin bug! Maybe you guys have an idea?

**Comment by PhilippS93** [comment]
Hello, I can confirm the bug! Everything went well until the update to iOS 15.0.

**Comment by bradmartin** [comment]
Any crash logs at all? Can you run the app in xcode and see? I have it on an app running on iOS15 and don't see a crash currently.

**Comment by bradmartin** [comment]
https://user-images.githubusercontent.com/6006148/137527262-77dd9fe4-ede1-40f9-8527-6359e3ea5a6d.mp4

**Comment by ghost** [comment]
I had the same thing happen running an ipad on 15 and an iphone on 14.8. The app and plugin worked in my simulators and when my iphone was attached as a test device. But would launch and immediately shut down when installed from Test Flight.

**Comment by bradmartin** [comment]
Ugh, those are harder to figure out for sure. I'll try to think of an app in release that might be using this and try to confirm. Even harder to track down if so. Maybe someone can provide a crash log from xcode organizer for a release build that has been distributed, which could help.

**Comment by ghost** [comment]
I don’t think that I can help there as my app is built using a script file which executes the following:

ns build ios --release --env.uglify --for-device --team-id <my team id>

And then uploads it to Test Flight with this command:

xcodebuild -exportArchive -allowProvisioningUpdates -exportOptionsPlist $PINFO_LIST -archivePath $ARCHIVE_DIR/$BUNDLE_ID.xcarchive -exportPath Release

Nachman Beinish Ross
On Oct 20, 2021, 12:24 PM -0400, Brad Martin ***@***.***>, wrote:


Ugh, those are harder to figure out for sure. I'll try to think of an app in release that might be using this and try to confirm. Even harder to track down if so. Maybe someone can provide a crash log from xcode organizer for a release build that has been distributed, which could help.

**Comment by bradmartin** [comment]
If you're authenticated with xcode, I think it should still show the distributed apps (but maybe not if all done with CLI). The logs usually take a day or two to show up from apple if there are crashes for specific builds. Just not fun to track down 🙃

**Comment by ghost** [comment]
I took a look at organizer in Xcode and it shows the versions that I released to Xcode but it doesn’t show any crash logs. I’ll look again in the next few days and let you know if anything shows up.

I’m really not an iOS or MacOS maven but I’m willing to try different things if instructions are sent. I’ve spent over 40 years developing and designing systems on DOS and then Windows. I retired three years ago and because I was going stir crazy after one day, I formed an LLC and started developing apps using Cordova and then quickly switched to Nativescript. I was still developing on a Windows laptop and then about 2 years ago I switched to a Mac. So while I’ve become proficient with MacOS, I only use Xcode to verify things like Signing & Capabilities. 

Nachman Beinish Ross
On Oct 20, 2021, 12:46 PM -0400, Brad Martin ***@***.***>, wrote:


If you're authenticated with xcode, I think it should still show the distributed apps (but maybe not if all done with CLI). The logs usually take a day or two to show up from apple if there are crashes for specific builds. Just not fun to track down 🙃

**Comment by bradmartin** [comment]
No worries, thanks for offering the help :) If you do see any crash logs show up in the organizer and want to share, just upload them on this issue and maybe it will point us in the right direction. Thanks.
